crates/cli-support/src/js/mod.rs

```diff
@@ -904,7 +904,11 @@ __wbg_set_wasm(wasm);"
                                 return await WebAssembly.instantiateStreaming(module, imports);
 
                             }} catch (e) {{
-                                if (module.headers.get('Content-Type') != 'application/wasm') {{
+                                const expectedTypes = new Set(['basic', 'cors', 'default']);
+
+                                const validResponse = module.ok && expectedTypes.has(module.type);
```

Comment:
Do we really need the expectedTypes as well here, or is the `module.ok` enough?

Reply:
My main annoyance is indeed with failed requests, but when I actually read the [spec](https://webassembly.github.io/spec/web-api/#compile-a-potential-webassembly-response), I also found another possibility:

> * If response is not [CORS-same-origin](https://html.spec.whatwg.org/#cors-same-origin), [reject](https://webidl.spec.whatwg.org/#reject) returnValue with a [TypeError](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror) and abort these substeps.

Reply:
Right, there are no headers in this case too. Okay, in that case, let's at least make sure to hoist the set population though out of the inner body to be statically defined. Alternatively just split the statement into three string equality checks.